dispatches/models/fossil_case/properties/thermaloil_properties.py

```diff
@@ -0,0 +1,430 @@
+
+        init_log.info('Initialization Step 1 Complete.')
+
+        # ---------------------------------------------------------------------
+        if state_vars_fixed is False:
+            # release state vars fixed during initialization if control
+            # volume didn't fix the state vars
+            if hold_state is True:
+                return flags
+            else:
+                blk.release_state(flags)
+        init_log.info('Initialization Complete.')
+
+    def release_state(blk, flags, outlvl=idaeslog.NOTSET):
+        # Method to release states only if explicitly called
+
+        if flags is None:
+            return
+        # Unfix state variables
+        revert_state_vars(blk, flags)
+        init_log.info('State Released.')
+
+
+@declare_process_block_class("ThermalOilStateBlock",
+                             block_class=_StateBlock)
+class ThermalOilStateBlockData(StateBlockData):
+    """
+    Property package for Thermal oil properties
+    """
+
+    def build(self):
+        """
+        Callable method for Block construction
+        """
+        super(ThermalOilStateBlockData, self).build()
+
+        # Create state variables
+        self._make_state_vars()
+
+        # Create required properties
+        self._make_prop_vars()
+
+        # Create constraints
+        self._make_constraints()
+
+    def _make_state_vars(self):
+        """Declare the necessary state variable objects."""
+        self.flow_mass = Var(initialize=100.0,
+                             domain=Reals,
+                             doc="Total mass flow [Kg/s]",
+                             units=pyunits.kg/pyunits.s)
+        self.temperature = Var(initialize=523,
+                               domain=Reals,
+                               doc="Temperature of thermal oil [K]",
+                               bounds=(260, 616),
+                               units=pyunits.K)
+        self.pressure = Var(initialize=101325,
+                            domain=Reals,
+                            doc="Pressure [Pa]",
+                            units=pyunits.Pa)
+
+    def _make_prop_vars(self):
+        """Make additional variables for calculations."""
+
+        self.enthalpy_mass = Var(self.phase_list,
+                                 initialize=1,
+                                 units=pyunits.J/pyunits.kg,
+                                 doc='Specific Enthalpy')
+
+    def _make_constraints(self):
+        """Create property constraints."""
+
+        # Specific heat capacity
+        self.cp_mass = Expression(
+            self.phase_list,
+            expr=((self.params.cp_param_2 * (self.temperature - self.params.ref_temperature) +
+                   self.params.cp_param_3 * (self.temperature - self.params.ref_temperature)**2 +
+                   self.params.cp_param_1)),
+            doc="Specific heat capacity [J/kg/K]"
+        )
+
+        # Specific Enthalpy
+        def enthalpy_correlation(self, p):
```

Comment:
Will there be a single phase always? If not, `enthalpy_mass` evaluates to the same value for all phases.

Reply:
Yes, the fluid will always be in the liquid phase. The property correlations and the bounds included are valid for liquid phase only. Also, we only add LiquidPhase from the property base. The phase index is kept in the model for consistency.